### PR TITLE
Tests: Fail on FutureWarning

### PR DESCRIPTION
**Pull request by akihironitta**
~Meant to be targeted to my fork...~

## What does this PR do?
Addresses: https://github.com/PyTorchLightning/pytorch-lightning/pull/11527#discussion_r787897493

### Does your PR introduce any breaking changes? If yes, please list them.
None

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [n/a] Did you make sure to **update the documentation** with your changes? (if necessary)
- [n/a] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [n/a] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Review by carmocca — APPROVED** [review]
It's good that you targeted master, this is a worthy addition even if no warnings are currently raised.

File: tests/benchmarks/test_basic_parity.py [hunk 1]
```diff
@@ -97,7 +97,6 @@ def measure_loops(cls_model, kind, num_runs=10, num_epochs=10):
         gc.collect()
         if device_type == "cuda":
             torch.cuda.empty_cache()
-            torch.cuda.reset_max_memory_cached()
```
Comment (akihironitta) [inline tests/benchmarks/test_basic_parity.py:100]: `reset_max_memory_cached()` only calls `reset_peak_memory_stats()` inside, which is not needed as we call the latter below.

You can check the impl. here: https://github.com/pytorch/pytorch/blob/a0ada2d22b0e612dbf50fa5e94e6454713af8bb4/torch/cuda/memory.py#L278

Comment (akihironitta) [reply tests/benchmarks/test_basic_parity.py:100]: Same applies to `reset_max_memory_allocated`.

https://github.com/pytorch/pytorch/blob/a0ada2d22b0e612dbf50fa5e94e6454713af8bb4/torch/cuda/memory.py#L252

File: setup.cfg [hunk 2]
```diff
@@ -27,6 +27,7 @@ addopts =
 filterwarnings =
     # error out on our deprecation warnings - ensures the code and tests are kept up-to-date
     error::pytorch_lightning.utilities.warnings.LightningDeprecationWarning
+    error::FutureWarning
```
Comment (rohitgr7) [inline setup.cfg:30]: will it start popping up test failures in tests in case any of the dependencies start raising this error internally?

Comment (carmocca) [reply setup.cfg:30]: Yes. But this is good to allow us to be on top of user-facing changes to those dependencies.

Comment (rohitgr7) [reply setup.cfg:30]: okay.. but still we can't say the same for dependencies of a dependency.
CI looks green now, but there is a possibility we might end up removing that in the future if that happens.

I hope we are not removing `LightningDeprecationWarning`?

Comment (carmocca) [reply setup.cfg:30]: We can ignore here warnings for specific modules or libraries if that happens.

**Comment by akihironitta** [comment]
I'll check the failing ones in the benchmark tests.

**Review by rohitgr7 — APPROVED** [review]

**Review by ethanwharris — APPROVED** [review]
LGTM :smiley:
